src/pods/toolbar/components/export-button/export-button.business.ts

```diff
@@ -28,6 +28,78 @@ export const calculateTableEndYPosition = (table: TableVm): number => {
   return table.y + tableHeight;
 };
 
+export const isTableOverlap = (table: TableVm, tables: TableVm[]): boolean => {
+  if (tables.length === 0) {
+    return false;
+  }
+
+  const expandedFields = expandTableFields(table.fields);
+  const tableEndXPosition = table.x + TABLE_CONST.TABLE_WIDTH;
+  const tableEndYPosition = calculateTableEndYPosition({
+    ...table,
+    fields: expandedFields,
+  });
+
```

Comment:
Let's do this in two steps and simplify the some (it will be two loops but since we will have less than 200 tables is worth having the code simples than going for a performance optimization).

```ts
const tablesMinusOriginTableCollection = tables.filter(t => t.id !== table.id);
return tablesMinusOriginTableCollection.some(t => {
```

Reply:
done